File: PWD007/example-omp.c
// PWD007: Unprotected multithreading recurrence
// https://www.codee.com/knowledge/pwd007

void foo() {
    int x[5], y[5];

    y[0] = 0;
    #pragma omp parallel for
    for (int i=1; i<5; i++) {
        y[i] = y[i-1] + x[i-1];
    }
}

File: PWD007/example-omp.f90
! PWD007: Unprotected multithreading recurrence
! https://www.codee.com/catalog/pwd007

SUBROUTINE foo
    IMPLICIT NONE
    INTEGER :: i, x(5), y(5)

    y(1)=0
    !$omp parallel do
    DO i = 2, 5
        y(i) = y(i-1) + x(i-1)
    END DO
END SUBROUTINE
